app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanel.tsx

```diff
@@ -45,57 +44,30 @@ export const addConnectedState = (servers: JenkinsServer[]): JenkinsServer[] =>
 		});
 };
 
-const addConnectedStateToSingleServer = (singleServer: JenkinsServer, allServers: JenkinsServer[]): JenkinsServer => {
```

Comment:
Hoisted this logic up one level so we can better set and render the updated servers.